static/js/services/target-generator.js

```diff
@@ -54,10 +54,12 @@ var moment = require('moment'),
           // unconfigured target type
           return;
         }
-        if (instance.deleted) {
-          delete target.instances[instance._id];
-        } else if (isRelevant(instance)) {
+        if (isRelevant(instance) && !instance.deleted) {
```

Comment:
Probably cheaper to check `.deleted` before calling `isRelevant()`, but anyway if an instance is deleted, it's presumably not relevant.  So I'd expect `isRelevant()` to perform the deleted check.